nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/ExecuteSQL.java

```diff
@@ -278,8 +289,18 @@ public void onTrigger(final ProcessContext context, final ProcessSession session
                 } else {
                     fileToProcess = session.write(fileToProcess, JdbcCommon::createEmptyAvroStream);
 
+                    fileToProcess = session.putAttribute(fileToProcess, CoreAttributes.MIME_TYPE.key(), JdbcCommon.MIME_TYPE_AVRO_BINARY);
                     session.transfer(fileToProcess, REL_SUCCESS);
                 }
+            } else if(resultCount == 0){
+                //If we had no inbound FlowFile, no exceptions, and the SQL generated no result sets (Insert/Update/Delete statements only)
+                // Then generate an empty Output FlowFile
+                FlowFile resultSetFF = session.create();
+
+                resultSetFF = session.write(resultSetFF, out -> JdbcCommon.createEmptyAvroStream(out));
+
+                resultSetFF = session.putAttribute(resultSetFF, CoreAttributes.MIME_TYPE.key(), JdbcCommon.MIME_TYPE_AVRO_BINARY);
```

Comment:
Should have `RESULT_ROW_COUNT` on it to tell downstream processors that it's got not rows in it.